ckan/types/__init__.py

```diff
@@ -69,13 +70,12 @@ class Context(TypedDict, total=False):
     here.
     """
     user: str
-    model: Model
     session: AlchemySession
 
     __auth_user_obj_checked: bool
     __auth_audit: list[tuple[str, int]]
-    auth_user_obj: Union["Model.User", "Model.AnonymousUser", None]
-    user_obj: "Model.User"
+    auth_user_obj: model.User | model.AnonymousUser | None
+    user_obj: model.User
```

Comment:
```suggestion
    user_obj: model.User
```

Shouldn't user_obj be as well: `model.User | model.AnonymousUser | None?

Reply:
No, seems like this never happens. `user_obj` is not a popular option and seems that only few actions are using it as some [sort of cache](https://github.com/ckan/ckan/blob/master/ckan/logic/action/update.py#L920-L923)

```py
    user_obj = model.User.get(id)
    if user_obj is None:
        raise NotFound('User was not found.')
    context['user_obj'] = user_obj
```

And as it never takes value of `current_user`, it never can be an `AnonymousUser` or `None`. At least not in existing CKAN code and there is no reason to allow abusing this property by extending it's type range.

As for `auth_user_obj`, it often [set by views](https://github.com/ckan/ckan/blob/master/ckan/views/dataset.py#L251) and takes value of the `current_user`, that's why it has to accept all corresponding types:

```py
    context: Context = {
        u'user': current_user.name,
        u'for_view': True,
        u'auth_user_obj': current_user
    }
```